combatMenuFunctions.py

```diff
@@ -1,70 +1,46 @@
-                classes.Player.__setattr__({
-                    classes.Player.locationTag,"locationTag",
-                    classes.World.world_map[horizontal][vertical]})
-
-def moveUp(): 
-    classes.Player.positionY -= 1 
-    if classes.Player.positionX in gridList and classes.Player.positionY < 0: 
-        classes.Player.positionY += 1
-        classes.Player.day -= 1
-        errorMessages.exitGridError()
-        setLocation() 
-    return (classes.Player.positionY, classes.Player.positionY+1)
-
-def moveDown():
-    classes.Player.positionY += 1
-    if  classes.Player.positionX in gridList and classes.Player.positionY > 7: 
-        classes.Player.positionY += 1
-        classes.Player.day -= 1
-        errorMessages.exitGridError()
-        setLocation() 
-    return (classes.Player.positionY, classes.Player.positionY-1)
-
-def moveLeft():
-    classes.Player.positionX -= 1
-    if  classes.Player.positionY in gridList and classes.Player.positionX < 0:
-        classes.Player.positionY += 1
-        classes.Player.day -= 1
-        errorMessages.exitGridError()
-        setLocation() 
-    return (classes.Player.positionX, classes.Player.positionX+1)
-
-def moveRight():
-    classes.Player.positionX += 1
-    if  classes.Player.positionX in gridList and classes.player.positionY > 7:
-        classes.Player.positionY += 1
-        classes.Player.day -= 1
-        errorMessages.exitGridError()
-        setLocation() 
-    return (classes.Player.positionX, classes.Player.positionX-1)
-
-def UserMovementOption():
+def UserMovementOption(hero, world):
     inputOption = input("Your Move:")
     if inputOption.isalpha:
         inputOption = inputOption.upper()
+        
         if inputOption == 'W':
-            moveUp()
-        if inputOption == 'A':
-            moveLeft()
-        if inputOption == 'S':
-            moveDown()
-        if inputOption == 'D':
-            moveRight()
-
+            hero.positionX -= 1
+            if hero.positionX < 0:
+                hero.positionX += 1
+                errorMessages.exitGridError()
+            else:
+                world.day += 1
+                
+        elif inputOption == 'A':
+            hero.positionY -= 1
+            if hero.positionY < 0:
+                hero.positionY += 1
+                errorMessages.exitGridError()
+            else:
+                world.day += 1                
+            
+        elif inputOption == 'S':
+            hero.positionX += 1
+            if hero.positionX > 7: 
+                hero.positionX -= 1
+            else:
+                world.day += 1
+            
+        elif inputOption == 'D':
+            hero.positionY += 1
+            if hero.positionY > 7:
+                hero.positionY -= 1
+            else:
+                world.day += 1
+                
 def attack():
     Rat = classes.Rat()
-    Player = classes.Player()
+    Player = hero()
```

Comment:
why u change dis part. There is no class named hero()

Reply:
I used replace all to replace all instances of the old statements used to reference the player class, so this ended up being changed and I missed it.
Since there are no rats in the game yet, no testing could be done and hence this issue wasn't found.

Reply:
Ok Please resolve this convo n push a commit to fix that line :)

Reply:
I have reverted it to what it originally was, but I'm not sure if it is worked. Testing will need to be conducted.